### PR TITLE
A/B Test Adding G Suite SKUs Select to CTA

### PR DESCRIPTION
**Pull request by cbauerman**
#### Changes proposed in this Pull Request

* Add Business Option to G Suite Purchase CTA AB Test
* Add new `/email/:domain/gsuite/new/:planType/:siteSlug` path
* Move `/email/:domain/gsuite-add-users/:siteSlug` to `/email/:domain/gsuite/add-users/:siteSlug` 
* Refactoring the G Suite Purchase CTA
* Added Tracks events for viewing the CTA and clicking the "Get G Suite X" button
* Removed G Suite Purchase CTA Snapshot tests now that component has increased connection to Redux state

**Wide**
Wide Example

**Mobile**
Mobile Example

**"Unlimited" Fine Print**
Screen Shot 2019-04-08 at 11 47 01 AM

#### Testing instructions

1. Enable Analytics logging by typing `localStorage.setItem( 'debug', 'calypso:analytics:*' )` into the console of your browser dev tools
1. Open the "Bug" Menu and confirm you are in the "hide" group of the `gSuiteSkuSelect` AB Test Screen Shot 2019-04-05 at 2 50 29 PM
1. Navigate to `/email/:domain/manage/:siteSlug` for a domain and site without G Suite
1. Confirm what you see matches this screen shot: Screen Shot 2019-04-05 at 2 55 18 PM
1. Make sure your Dev Tools are open so you can confirm Tracks events are correct ( clearing it would probably help )
1. Click the "Add G Suite" Button
1. Confirm you have been taken to the `Add G Suite` Page: Screen Shot 2019-04-05 at 2 58 37 PM
1. Inspect the url and confirm it matches `/email/:domain/gsuite/new/basic/:siteSlug`
1. Check your Dev tools console and verify a `calypso_email_gsuite_purchase_cta_get_gsuite_click` event was recorded, with the properties `plan_type: 'basic'` and `domain` for the domain
1. Add G Suite user info and continue to the cart
1. Confirm that the item in the cart is regular "G Suite"
1. Don't complete the purchase, remove all items from cart
1. Switch into the "show" group of the `gSuiteSkuSelect` AB Test
1. Navigate back to `/email/:domain/manage/:siteSlug`
1.  Confirm what you see matches this screen shot: Screen Shot 2019-04-05 at 3 07 27 PM
1. Click the "Add G Suite Business" Button
1. Confirm you have been taken to the `Add G Suite` Page
1. Inspect the url and confirm it matches `/email/:domain/gsuite/new/business/:siteSlug`
1. Check your Dev tools console and verify a `calypso_email_gsuite_purchase_cta_get_gsuite_click` event was recorded, with the properties `plan_type: 'business'` and `domain` for the domain
1. Add G Suite user info and continue to the cart
1. Confirm that the item in the cart is "G Suite with Unlimited Storage" and the price is correct
1. Don't complete the purchase, remove all items from cart
1. Navigate to `/email/:domain/manage/:siteSlug` for a site & domain with G Suite
1. Click the "Add G Suite User" button
1. Confirm you have been taken to the `Add G Suite` Page
1. Inspect the url and confirm it matches `/email/:domain/gsuite/add-users/:siteSlug`
1. Change the url to `/email/:domain/add-gsuite-users/:siteSlug`
1. Confirm you are redirected to the same page
1. Add G Suite User info and continue to cart
1. Confirm the item in the cart is "Extra License for G Suite"

### CODE REVIEW

**Comment by cbauerman** [comment]
This PR needs 2 more things before it is fully ready for review:

1. The old ‘add-gsuite-users’ route needs a legacy redirect to the new ‘gsuite/add’
1. An event when the CTA is loaded

File: client/my-sites/email/gsuite-purchase-features/index.jsx [hunk 1]
```diff
@@ -27,10 +27,15 @@ class GSuitePurchaseFeatures extends React.Component {
 
 	getStorageText() {
 		const { productSlug, translate } = this.props;
-		if ( 'gapps' === productSlug ) {
-			return translate( 'Get 30GB of storage for all your files synced across devices.' );
-		} else if ( 'gapps_unlimited' === productSlug ) {
-			return translate( 'Get unlimited storage for all your files synced across devices.' );
+		switch ( productSlug ) {
```
Comment (belcherj) [inline client/my-sites/email/gsuite-purchase-features/index.jsx]: Why switch to a case statements, will it ever not equal 'gapps' or 'gapps_unlimited'

File: client/my-sites/email/gsuite-purchase-features/index.jsx [hunk 2]
```diff
@@ -124,7 +129,7 @@ const learnMoreClick = domainName =>
 
 GSuitePurchaseFeatures.propTypes = {
 	domainName: PropTypes.string.isRequired,
-	productSlug: PropTypes.string.isRequired,
+	productSlug: PropTypes.string,
```
Comment (belcherj) [inline client/my-sites/email/gsuite-purchase-features/index.jsx]: Do we not need a slug?

File: client/my-sites/email/gsuite-purchase-features/index.jsx (hunk 1 above)
Comment (cbauerman) [reply client/my-sites/email/gsuite-purchase-features/index.jsx]: The `default` case is used is used in the A/B test to show "30GB or unlimited" to imply the amount of storage is dependent on the sku chosen. The linter was giving me a really hard time about using `if ... else if ... else` with each having a return, so I just switched to a case statement.

File: client/my-sites/email/gsuite-purchase-features/index.jsx (hunk 2 above)
Comment (cbauerman) [reply client/my-sites/email/gsuite-purchase-features/index.jsx]: nope, not passing in a productSlug gets the plan-agnostic "30GB or unlimited storage" mentioned above

File: client/my-sites/email/gsuite-purchase-cta/test/index.js [hunk 3]
```diff
@@ -1,62 +0,0 @@
-/** @format */
```
[the remark this answers is not in the document]
Comment (cbauerman) [reply client/my-sites/email/gsuite-purchase-cta/test/index.js]: I was being a little lazy 😖. I will give preserving them another try!

File: client/my-sites/email/gsuite-purchase-cta/index.jsx [hunk 4]
```diff
@@ -12,105 +13,127 @@ import React, { Fragment } from 'react';
...
+			null === basicAnnualPrice ? '-' : getAnnualPrice( basicAnnualPrice, currencyCode );
+		const businessAnnualPrice = get(
+			productsList,
+			[ 'gapps_unlimited', 'prices', currencyCode ],
+			null
+		);
+		const businessAnnualPriceFormatted =
+			null === businessAnnualPrice ? '-' : getAnnualPrice( businessAnnualPrice, currencyCode );
+
+		const inABTest = abtest( 'gSuiteSkuSelect' ) === 'show';
```
[the remark this answers is not in the document]
Comment (cbauerman) [reply client/my-sites/email/gsuite-purchase-cta/index.jsx]: Seems wise, `skuSelect` is a little hard to parse

File: client/my-sites/email/gsuite-purchase-cta/sku-info.jsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
+/** @format */
```
[the remark this answers is not in the document]
Comment (cbauerman) [reply client/my-sites/email/gsuite-purchase-cta/sku-info.jsx]: That is a good idea, esp. since the testing instructions have gotten a tad long.

File: client/my-sites/email/gsuite-purchase-cta/index.jsx [hunk 6]
```diff
@@ -12,105 +13,127 @@ import React, { Fragment } from 'react';
...
+			domain: this.props.domainName,
+		} );
 	}
 
 	renderCta() {
-		const { annualPrice, domainName, productSlug, translate } = this.props;
+		const { currencyCode, domainName, productsList, translate } = this.props;
 		const upgradeAvailable = config.isEnabled( 'upgrades/checkout' );
 
+		const basicAnnualPrice = get( productsList, [ 'gapps', 'prices', currencyCode ], null );
```
[the remark this answers is not in the document]
Comment (cbauerman) [reply client/my-sites/email/gsuite-purchase-cta/index.jsx]: It also might be a decent selector. I will give it some thought.

**Comment by gravityrail** [comment]
One UI note:

- Refer to regular "G Suite" as "G Suite Basic" in the UI. This increases the appeal of Business by implying that G Suite Basic is bare-bones.
